# Livro para consulta:
- https://jakevdp.github.io/PythonDataScienceHandbook/03.08-aggregation-and-grouping.html
- https://jakevdp.github.io/PythonDataScienceHandbook/03.09-pivot-tables.html
    

# 1. Importando bibliotecas <a name="import"></a>

<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [104]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline

# 2. Carregando o dataframe SINASC <a name="read"></a>
<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [105]:
sinasc_raw = pd.read_csv('SINASC_RO_2019.csv')
sinasc_raw.head()

ORIGEM   CODESTAB  CODMUNNASC  LOCNASC  IDADEMAE  ESTCIVMAE  \
0       1  2679477.0      110001        1        19        5.0   
1       1  2679477.0      110001        1        29        2.0   
2       1  2679477.0      110001        1        37        9.0   
3       1  2516500.0      110001        1        30        5.0   
4       1  2516500.0      110001        1        30        2.0   

            ESCMAE  CODOCUPMAE  QTDFILVIVO  QTDFILMORT  ...  KOTELCHUCK  \
0      8 a 11 anos         NaN         0.0         0.0  ...           5   
1      8 a 11 anos    999992.0         1.0         0.0  ...           5   
2      8 a 11 anos    513205.0         2.0         0.0  ...           5   
3  12 anos ou mais    231205.0         0.0         0.0  ...           4   
4      8 a 11 anos    999992.0         1.0         0.0  ...           5   

  CONTADOR munResStatus munResTipo               munResNome  munResUf  \
0        1        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
1        2        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
2        3        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
3        4        ATIVO      MUNIC  Alto Alegre dos Parecis  Rondônia   
4        5        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   

   munResLat munResLon  munResAlt  munResArea  
0  -11.93554 -61.99982      338.0    7067.025  
1  -11.93554 -61.99982      338.0    7067.025  
2  -11.93554 -61.99982      338.0    7067.025  
3  -12.13178 -61.85308      397.0    3958.273  
4  -11.93554 -61.99982      338.0    7067.025  

[5 rows x 69 columns]

# Tarefa 1

### 1. Idade media das mães e dos pais por município (coluna munResNome)


In [132]:
sinasc_raw.groupby('munResNome')[['IDADEMAE', 'IDADEPAI']].mean().head()

IDADEMAE   IDADEPAI
munResNome                                   
Alta Floresta D'Oeste    25.991826  29.548701
Alto Alegre dos Parecis  24.844156  29.179104
Alto Paraíso             24.960177  28.833333
Alvorada D'Oeste         25.770732  30.757282
Ariquemes                25.607866  32.466667

In [150]:
#ALTERNATIVA COM O 'pivot_table':

sinasc_raw.pivot_table(values = ['IDADEMAE', 'IDADEPAI'], index = 'munResNome', aggfunc='mean').head()

IDADEMAE   IDADEPAI
munResNome                                   
Alta Floresta D'Oeste    25.991826  29.548701
Alto Alegre dos Parecis  24.844156  29.179104
Alto Paraíso             24.960177  28.833333
Alvorada D'Oeste         25.770732  30.757282
Ariquemes                25.607866  32.466667

### 2. Peso médio dos bebes por sexo que nasceram no dia do seu aniversário por faixas de escolaridade mae
Ex: Você, aluna(o), nasceu no dia 10/01, então você precisa filtrar o conjunto de dados nessa data e calcular o peso médio dos bebês de cada sexo por faixa de escolaridade da mãe.

In [124]:
# peso médio dos bebês que nasceram em 28/06, por sexo:

datanasc = sinasc_raw.loc[(sinasc_raw['DTNASC'] == '2019-06-28')]
datanasc.pivot_table(values = ['PESO'], index = 'SEXO', aggfunc='mean')

PESO
SEXO                  
Feminino   3187.675000
Masculino  3236.395349

### 3. Qual o municipio que nasceu menos bebe em 2019?
    - qual a idade media, maxima, minima das maes nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?

In [109]:
# R: Município de Castanheiras.

resultado = sinasc_raw.groupby('munResNome')['DTNASC'].count().sort_values().head()
resultado

munResNome
Município ignorado - RO     1
Castanheiras               32
Pimenteiras do Oeste       40
Primavera de Rondônia      43
Parecis                    44
Name: DTNASC, dtype: int64

In [110]:
# qual a idade média das mães nesse municipio?
# qual a idade média dos pais nesse municipio?

mun_Castanheiras = sinasc_raw.loc[(sinasc_raw['munResNome'] == 'Castanheiras')]
mun_Castanheiras[['IDADEMAE', 'IDADEPAI']].mean()

IDADEMAE    27.281250
IDADEPAI    30.392857
dtype: float64

In [111]:
# qual a idade máxima das mães nesse municipio?
# qual a idade máxina dos pais nesse municipio?

mun_Castanheiras[['IDADEMAE', 'IDADEPAI']].max()

IDADEMAE    39.0
IDADEPAI    43.0
dtype: float64

In [122]:
# qual a idade mínima das mães nesse municipio?
# qual a idade mínima dos pais nesse municipio?

mun_Castanheiras[['IDADEMAE', 'IDADEPAI']].min()

IDADEMAE    17.0
IDADEPAI    17.0
dtype: float64

In [130]:
# ALTERNATIVA:

# qual a idade media, maxima, minima das maes nesse municipio?
# qual a idade media, maxima, minima dos pais nesse municipio

mun_Castanheiras[['IDADEMAE', 'IDADEPAI']].agg({'IDADEMAE':['mean','max','min'], 
                                                'IDADEPAI':['mean','max','min']})

IDADEMAE   IDADEPAI
mean  27.28125  30.392857
max   39.00000  43.000000
min   17.00000  17.000000

### 4. Qual o municipio que nasceu mais bebe no mês de março?
    - qual a quantidade de filhos vivos media, maxima, minima nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?



In [114]:
# R: Município de Porto Velho.
    
sinasc_raw.loc[(sinasc_raw['DTNASC'] > '2019-02-28') & (sinasc_raw['DTNASC'] < '2019-04-01')]
datanasc_marco = sinasc_raw.groupby('munResNome')['DTNASC'].count().sort_values(ascending=False)
datanasc_marco.head()

munResNome
Porto Velho    8437
Ji-Paraná      2182
Ariquemes      1729
Vilhena        1590
Cacoal         1374
Name: DTNASC, dtype: int64

In [115]:
mun_PortoVelho = sinasc_raw.loc[(sinasc_raw['munResNome'] == 'Porto Velho')]
mun_PortoVelho.head()

ORIGEM   CODESTAB  CODMUNNASC  LOCNASC  IDADEMAE  ESTCIVMAE       ESCMAE  \
123       1  2515520.0      110002        1        22        1.0  8 a 11 anos   
201       1  2494299.0      110002        1        22        2.0  8 a 11 anos   
246       1  2494299.0      110002        1        29        1.0  8 a 11 anos   
272       1  2494299.0      110002        1        27        2.0   4 a 7 anos   
355       1  2515504.0      110002        1        33        2.0  8 a 11 anos   

     CODOCUPMAE  QTDFILVIVO  QTDFILMORT  ...  KOTELCHUCK CONTADOR  \
123    999992.0         0.0         0.0  ...           5      125   
201    999992.0         0.0         0.0  ...           5      204   
246    999992.0         2.0         0.0  ...           3      249   
272    999992.0         2.0         1.0  ...           5      275   
355    999992.0         0.0         0.0  ...           5      359   

    munResStatus munResTipo   munResNome  munResUf  munResLat munResLon  \
123        ATIVO      MUNIC  Porto Velho  Rondônia   -8.76889 -63.83145   
201        ATIVO      MUNIC  Porto Velho  Rondônia   -8.76889 -63.83145   
246        ATIVO      MUNIC  Porto Velho  Rondônia   -8.76889 -63.83145   
272        ATIVO      MUNIC  Porto Velho  Rondônia   -8.76889 -63.83145   
355        ATIVO      MUNIC  Porto Velho  Rondônia   -8.76889 -63.83145   

     munResAlt  munResArea  
123       87.0   34096.394  
201       87.0   34096.394  
246       87.0   34096.394  
272       87.0   34096.394  
355       87.0   34096.394  

[5 rows x 69 columns]

In [129]:
# qual a quantidade de filhos vivos media, maxima, minima nesse municipio?


mun_PortoVelho[['QTDFILVIVO']].agg({'QTDFILVIVO':['mean','max','min']})

QTDFILVIVO
mean    1.167236
max    12.000000
min     0.000000

In [128]:
# qual a idade media, maxima, minima dos pais nesse municipio?

mun_PortoVelho[['IDADEMAE', 'IDADEPAI']].agg({'IDADEMAE':['mean','max','min'],
                                              'IDADEPAI':['mean','max','min']})

IDADEMAE   IDADEPAI
mean  26.31267  32.352679
max   47.00000  65.000000
min   12.00000  16.000000

### Analise as respostas encontradas, tire algum insight delas, conte pra gente algo encontrado nos dados. Algo que você julgue relevante e novo pra você.

Exemplo:
- Ah, descobri que a idade mediana das mulheres que deram a luz no ano de 2019 dos municipios x é maior que y.

Dos dados analisados é possível identificar:

- que a idade média das mães para o município de Porto Velho é de 26.31267 anos, enquanto que no município de Castanheiras é de 27.28125 anos. Logo, a idade média das mães é maior no muncípio de Castanheiras;

- que o peso médio de bebês do sexo masculino para o ano de 2019 no Estado de RO foi maior do que o dos bebês do sexo feminino;

- que a idade mínima das mães e dos pais no município de Porto Velho é menor que no município de Castanheiras;

- que a idade máxima das máes e dos pais no município de Castanheiras é menor que no município de Porto velho.

